questions.php

```diff
@@ -187,7 +187,7 @@
                 // just remove the dependency and inform the user about it.
                 $haschildren = $questionnaire->get_all_dependants($qid);
             }
-            if (count($haschildren) != 0) {
+            if ($haschildren !== null and (count($haschildren->directs) + count($haschildren->indirects)) > 0) {
```

Comment:
The block starting at "if ($questionnairehasdependencies) {" can be simplified down to:

```
            $action = "confirmdelquestion";
            if ($questionnairehasdependencies) {
                // Important: due to possibly multiple parents per question
                // just remove the dependency and inform the user about it.
                $haschildren = $questionnaire->get_all_dependants($qid);
                if (count($haschildren->directs) + count($haschildren->indirects) > 0) {
                    $action = "confirmdelquestionparent";
                }
            }

```
The $haschildren will always be a stdClass with "directs" and "indirects" properties, so we can just check the "questionnairehasdependencies" boolean to determine the rest.

Reply:
This was already implemented in MOODLE_400_STABLE as part of https://github.com/PoetOS/moodle-mod_questionnaire/pull/442. I will backport it.